docs/earthfile/earthfile.md

```diff
@@ -1143,6 +1143,58 @@ Same as [`RUN --mount <mount-spec>`](#mount-less-than-mount-spec-greater-than).
 
 Same as [`RUN --secret <env-var>=<secret-ref>`](#secret-less-than-env-var-greater-than-less-than-secret-ref-greater-than).
 
+## WAIT
+
+Enable via `VERSION --wait-block 0.6`.
+
+#### Synopsis
+
+* ```
+  WAIT
+    <wait-block>
+  END
+  ```
+
+#### Description
+
+The `WAIT` clause will execute `BUILD`, and `SAVE IMAGE` commands contained in the `<wait-block>` in parallel, and wait for them to complete.
+
+Commands containing a `--push` (e.g. `RUN --push ...`, `SAVE IMAGE --push ...`), as well as `SAVE ARTIFACT ... AS LOCAL` commands will be performed as encountered; previously (when no the `--wait-block` feature is not enabled),
+push-based commands were batched together and executed, in a seperate phase, after all non-push commands were successfully executed.
```

Comment:
I took a stab to provide a simpler explanation. WDYT?

```suggestion
The `WAIT` clause will execute all the commands contained within and wait for them to complete, including producing executing pushes and producing local outputs, before continuing. This can be used to control the order of interactions with the outside world.

Even though the `WAIT` clause limits parallelism by forcing everything within it to finish executing before continuing, the commands **within** a `WAIT` block execute with full parallelism.
```

Reply:
I like it! I reworded it slightly to make it more concise -- thoughts?

Reply:
Looks good!